### DISCUSSION TITLE
How to configure codecs when using Pythonnet in Python

### DISCUSSION THREAD

**Discussion by simeydk** [pythonnet/pythonnet#2395] [Q&A]
Hello 

# Background
I am a Python developer using Pythonnet to call a C# DLL from Python. My project currently does not contain any C# source code, and only calls a supplied pre-supplied C# library.

I am using Pythonnet to access `foo.dll` from Python. 
`foo.dll` is an external DLL that I do not have access to the source code to. 
`foo.dll` consists of many custom classes and methods. 
This makes it awkward to use the library from within Python, since I keep getting opaque class objects as return types. What is particularly cumbersome is figuring out call signatures for class methods.

My understanding is [codecs](https://pythonnet.github.io/pythonnet/codecs.html) are the solution to this problem. The docs mention codecs, and how some examples of how to implement codecs when calling Python from within C#. 

# My question
How can I use codecs (or something else?) to transform the C# classes into more ergonomic Python classes?

**Comment by DaveSkender** [comment]
Looks like you're getting all 🦗🦗🦗 and no ❤️ here.

Here's [some examples](https://github.com/facioquo/stock-indicators-python/blob/a4a39ab515472bd64e7aa2eadcabb25131f9934b/stock_indicators/_cslib/__init__.py#L28-L42) of how we'd imported the .NET classes in our Stock Indicators for Python library, which similarly wraps an uncoordinated .NET DLL.  We're not using a _codec_ for full class objects; at least, not in any proper encode/decode form.

One thing to note, the built-in Python.NET codecs don't support all data types, likely only primitives and some other simple types, and not `decimal`, so if you're seeing problem, that's likely the reason. I've asked a [similar question](https://github.com/pythonnet/pythonnet/discussions/2432).
